File: VB/Scaffolding.ValidationErrors/Common/DataModel/IUnitOfWork.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq

Namespace Scaffolding.ValidationErrors.Common.DataModel
	Public Interface IUnitOfWork
		Sub SaveChanges()
		Function GetState(ByVal entity As Object) As EntityState
		Sub Update(ByVal entity As Object)
		Sub Detach(ByVal entity As Object)
	End Interface
	Public Interface IUnitOfWorkFactory(Of TUnitOfWork As IUnitOfWork)
		Function CreateUnitOfWork() As TUnitOfWork
	End Interface
End Namespace

File: CS/Scaffolding.ValidationErrors/Common/DataModel/IUnitOfWork.cs
using System;
using System.Linq;

namespace Scaffolding.ValidationErrors.Common.DataModel {
    public interface IUnitOfWork {
        void SaveChanges();
        EntityState GetState(object entity);
        void Update(object entity);
        void Detach(object entity);
    }
    public interface IUnitOfWorkFactory<TUnitOfWork> where TUnitOfWork : IUnitOfWork {
        TUnitOfWork CreateUnitOfWork();
    }
}
